src/main/java/org/pethotel/HeavenForPets/repository/RoomRepository.java

```diff
@@ -33,4 +34,16 @@
 
     @Query("SELECT e FROM RoomEntity e WHERE e.petType is null")
     List<RoomEntity> findAllPlantRooms();
+
+    @Query("SELECT e FROM RoomEntity e WHERE e.petType is not null")
+    List<RoomEntity> findAllPetRooms();
+
+    @Query("SELECT count(e) FROM RoomEntity e WHERE e.petType is not null")
+    int countNumbersOfPetRooms();
+
+    @Query("SELECT count(e) FROM RoomEntity e WHERE e.petType is null")
+    int countNumbersOfPlantRooms();
+
+    @Query("SELECT e FROM RoomEntity e JOIN e.shelfEntities se where se.plantInsolation = :aa and se.free = '1' and (e.temperature between :minTemp and :maxTemp)")
+    List<RoomEntity> getAllRoomsWithProperShelves(@Param ("aa") PlantInsolation plantInsolation, @Param ("minTemp") int minTemp, @Param ("maxTemp") int maxTemp);
```

Comment:
"aa" is not good name, change to "pi" or "plantInsolation"

Reply:
done